### PR TITLE
Fix tapping on a test in flutter run

### PR DESCRIPTION
**Pull request by Hixie**
Also:

* Remove find.byIcon since it's identical to find.icon. (I sent mail to flutter-dev about this.)

* Fix IconData's operator== and hashCode, which had not been updated when we added fields.

* Add the byTooltip finder to the list of suggested finders.

* Make the suggested Key finder prettier.

### CODE REVIEW

**Comment by Hixie** [comment]
I can't figure out how to test the fix for the tapping bug that this fixes. It's in the part of the test framework code that explicitly runs for events coming from outside the test...

**Comment by Hixie** [comment]
Fixes https://github.com/flutter/flutter/issues/5188

File: packages/flutter_test/lib/src/widget_tester.dart [hunk 1]
```diff
@@ -324,9 +331,18 @@ class WidgetTester extends WidgetController implements HitTestDispatcher, Ticker
       int totalNumber = 0;
       debugPrint('Some possible finders for the widgets at ${binding.globalToLocal(event.position)}:');
       for (Element element in candidates) {
-        if (totalNumber > 10)
+        if (totalNumber > 13)
```
Comment (xster) [inline packages/flutter_test/lib/src/widget_tester.dart]: Maybe add a comment on how this number is derived?

**Review by xster — COMMENTED** [review]
LG

File: packages/flutter_test/lib/src/widget_tester.dart (hunk 1 above)
Comment (Hixie) [reply packages/flutter_test/lib/src/widget_tester.dart]: It's just whatever felt right to me...

Comment (Hixie) [reply packages/flutter_test/lib/src/widget_tester.dart]: added a comment

**Comment by Hixie** [comment]
Will land on green.

**Comment by tvolkert** [comment]
The removal of `find.icon` should be announced as a breaking change (a bunch of tests within Google had to be updated).

**Comment by Hixie** [comment]
it was

**Pull request merged**